File: src/main/java/com/asos/reservationSystem/domain/dto/UserDto.java
package com.asos.reservationSystem.domain.dto;

import com.asos.reservationSystem.domain.entities.Role;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserDto {
    private Long id;
    private String degree;
    private String firstName;
    private String lastName;
    private String email;
    private String phoneNumber;
    private Role role;
    private String description;
}

File: ASOS-Frontend/DTOs/Account/UserDto.cs
using System.Text.Json.Serialization;

namespace DTOs.Account
{
    public class UserDto
    {
        [JsonPropertyName("degree")]
        public string? Degree { get; set; }

        [JsonPropertyName("email")]
        public string Email { get; set; }

        [JsonPropertyName("firstName")]
        public string FirstName { get; set; }

        [JsonPropertyName("lastName")]
        public string LastName { get; set; }

        [JsonPropertyName("role")]
        public string Role { get; set; }

        [JsonPropertyName("phoneNumber")]
        public string PhoneNumber { get; set; }

        [JsonPropertyName("description")]
        public string? Description { get; set; }

        [JsonPropertyName("id")]
        public long Id { get; set; }
    }
}
